src/applications/vaos/utils/eligibility.js

```diff
@@ -37,90 +52,95 @@ export async function getEligibilityData(
   isDirectScheduleEnabled,
 ) {
   const facilityId = getFacilityIdFromLocation(location);
-  const eligibilityChecks = [
-    checkPastVisits(systemId, facilityId, typeOfCareId, 'request').catch(
-      createErrorHandler('request', 'request-check-past-visits-error'),
-    ),
-    getRequestLimits(facilityId, typeOfCareId).catch(
+  const directSchedulingAvailable =
+    location.legacyVAR.directSchedulingSupported && isDirectScheduleEnabled;
+
+  const eligibilityChecks = {
+    requestLimits: getRequestLimits(facilityId, typeOfCareId).catch(
       createErrorHandler(
         'request',
         'request-exceeded-outstanding-requests-error',
       ),
     ),
-  ];
+  };
 
-  if (location.legacyVAR.directSchedulingSupported && isDirectScheduleEnabled) {
-    eligibilityChecks.push(
-      checkPastVisits(systemId, facilityId, typeOfCareId, 'direct').catch(
-        createErrorHandler('direct', 'direct-check-past-visits-error'),
-      ),
-    );
-    eligibilityChecks.push(
-      getAvailableClinics(facilityId, typeOfCareId, systemId).catch(
-        createErrorHandler('direct', 'direct-available-clinics-error'),
-      ),
-    );
-    eligibilityChecks.push(
-      getLongTermAppointmentHistory().catch(
-        createErrorHandler('direct', 'direct-no-matching-past-clinics-error'),
-      ),
+  if (typeOfCareId !== PRIMARY_CARE) {
+    eligibilityChecks.requestPastVisit = checkPastVisits(
+      systemId,
+      facilityId,
+      typeOfCareId,
+      'request',
+    ).catch(createErrorHandler('request', 'request-check-past-visits-error'));
+  }
+
+  if (directSchedulingAvailable) {
+    if (typeOfCareId !== PRIMARY_CARE) {
+      eligibilityChecks.directPastVisit = checkPastVisits(
+        systemId,
+        facilityId,
+        typeOfCareId,
+        'direct',
+      ).catch(createErrorHandler('direct', 'direct-check-past-visits-error'));
+    }
+
+    eligibilityChecks.clinics = getAvailableClinics(
+      facilityId,
+      typeOfCareId,
+      systemId,
+    ).catch(createErrorHandler('direct', 'direct-available-clinics-error'));
+    eligibilityChecks.pastAppointments = getLongTermAppointmentHistory().catch(
+      createErrorHandler('direct', 'direct-no-matching-past-clinics-error'),
     );
   }
 
-  const [requestPastVisit, requestLimits, ...directData] = await Promise.all(
-    eligibilityChecks,
-  );
-  let eligibility = {
-    requestPastVisit,
-    requestLimits,
+  const results = await promiseAllFromObject(eligibilityChecks);
```

Comment:
Nice touch.  I think the `promiseAllFromtObject()` is much clearer to read than all the pushes we were doing to `eligibilityChecks`.

Reply:
Some of the older promise libraries would let you pass either an array or object to Promise.all, always thought that was a nice feature.